### PR TITLE
tests: attestation: Restrict sample policy use

### PR DESCRIPTION
**Pull request by stevenhorsman**
- We only want to enable the sample verifier in the KBS for non-TEE tests, so prevent an edge case where the TEE platform isn't set up correctly and we might fall back to the sample and get false positives. To prevent this we add guards around the sample policy enablement and only run it for non confidential hardware

### CODE REVIEW

**Review by fidencio — APPROVED** [review]
lgtm, thanks @stevenhorsman!

**Comment by fidencio** [comment]
Befoire merging, I'd like to run this locally later Today, just to avoid burning CI cycles.

File: tests/integration/kubernetes/k8s-confidential-attestation.bats [hunk 1]
```diff
@@ -64,7 +67,10 @@ setup() {
 }
 
 @test "Cannot get CDH resource when deny-all policy is set" {
-	kbs_set_deny_all_resources
+	if ! is_confidential_hardware; then
+		kbs_set_deny_all_resources
+	fi
```
Comment (fitzthum) [inline tests/integration/kubernetes/k8s-confidential-attestation.bats:72]: I think `kbs_set_deny_all_resources` should still be called for all platforms. Otherwise I don't think this test will pass.

Comment (fidencio) [reply tests/integration/kubernetes/k8s-confidential-attestation.bats:72]: Very good catch!

**Review by fitzthum — APPROVED** [review]
LGTM

**Comment by fidencio** [comment]
TDX failure:
```
#   Warning  Failed   2s (x8 over 82s)  kubelet  Error: failed to create containerd container: create instance 4879: object with key "4879" already exists: unknown`
```

I will re-run after the other tests finish, but we're hitting this more often than I'd like to see it happening.

**Review by wainersm — APPROVED** [review]
LGTM!

**Comment by wainersm** [comment]
I realized our confidential tests have too many boilerplates. I'm planning to refactor the tests to reduce that, but first wait the PR for registry authentication and sealed secrets land.

**Review by sprt — APPROVED** [review]

**Comment by fidencio** [comment]
TDX CDH test passed, I will merge it after the TDX CI finishes.
